system/jlib/jstats.cpp

```diff
@@ -775,190 +775,194 @@ class StatisticMeta
+    { NUMSTAT(Executions), "The number of times a graph has been executed" },
+    { TIMESTAT(TotalNested), UNUSED },
     { CYCLESTAT(LocalExecute) },
-    { NUMSTAT(Compares) },
-    { NUMSTAT(ScansPerRow) },
-    { NUMSTAT(Allocations) },
-    { NUMSTAT(AllocationScans) },
-    { NUMSTAT(DiskRetries) },
+    { NUMSTAT(Compares), UNUSED },
+    { NUMSTAT(ScansPerRow), UNUSED },
+    { NUMSTAT(Allocations), "The number of allocations from the row memory" },
+    { NUMSTAT(AllocationScans), "The number of scans within the memory manager when allocating row memory\nOnly applies to the scanning heap manager (not used by default)" },
+    { NUMSTAT(DiskRetries), "The number of times an io operation was retried.\nIf this is non zero it may suggest a problem with the underlying disk storage" },
     { CYCLESTAT(Elapsed) },
     { CYCLESTAT(Remaining) },
     { CYCLESTAT(Soapcall) },
     { CYCLESTAT(FirstExecute) },
     { CYCLESTAT(TotalNested) },
-    { TIMESTAT(Generate) },
+    { TIMESTAT(Generate), "Time taken to generate the c++ code from the parsed ECL" },
     { CYCLESTAT(Generate) },
-    { WHENFIRSTSTAT(Started) },
-    { WHENLASTSTAT(Finished) },
-    { NUMSTAT(AnalyseExprs) },
-    { NUMSTAT(TransformExprs) },
-    { NUMSTAT(UniqueAnalyseExprs) },
-    { NUMSTAT(UniqueTransformExprs) },
-    { NUMSTAT(DuplicateKeys) },
-    { NUMSTAT(AttribsProcessed) },
-    { NUMSTAT(AttribsSimplified) },
-    { NUMSTAT(AttribsFromCache) },
-    { NUMSTAT(SmartJoinDegradedToLocal) },
-    { NUMSTAT(SmartJoinSlavesDegradedToStd) },
-    { NUMSTAT(AttribsSimplifiedTooComplex) },
-    { NUMSTAT(SysContextSwitches) },
-    { TIMESTAT(OsUser) },
-    { TIMESTAT(OsSystem) },
-    { TIMESTAT(OsTotal) },
+    { WHENFIRSTSTAT(Started), "Time when this activity or operation started" },
+    { WHENLASTSTAT(Finished), "Time when this activity or operation finished" },
+    { NUMSTAT(AnalyseExprs), "Code generator internal.\nThe number of expressions that were processed by transformer::analyse()" },
+    { NUMSTAT(TransformExprs), "Code generator internal.\nThe number of expressions that were processed by transformer::transform()" },
+    { NUMSTAT(UniqueAnalyseExprs), "Code generator internal.\nThe number of unique expressions that were processed by transformer::analyse()" },
+    { NUMSTAT(UniqueTransformExprs), "Code generator internal.\nThe number of unique expressions that were processed by transformer::transform()" },
+    { NUMSTAT(DuplicateKeys), "The number of duplicate keys that were present in the index" },
+    { NUMSTAT(AttribsProcessed), "The number of attributes processed when parsing the ECL" },
+    { NUMSTAT(AttribsSimplified), UNUSED },
+    { NUMSTAT(AttribsFromCache), UNUSED },
+    { NUMSTAT(SmartJoinDegradedToLocal), "The number of times a global smart-join switched to a LOCAL JOIN (with distribute)" },
+    { NUMSTAT(SmartJoinSlavesDegradedToStd), "The number of times a global smart-join degraded to a standard join" },
```

Comment:
is it worth saying, after newline. That the above 2 stats will either be 0 or 1, unless in a LOOP?